### PR TITLE
fix clippy warnings

### PR DESCRIPTION
**Pull request by jhscheer**

### CODE REVIEW

File: src/uu/test/src/parser.rs [hunk 1]
```diff
@@ -121,11 +121,7 @@ impl Parser {
     /// Test if the next token in the stream is a BOOLOP (-a or -o), without
     /// removing the token from the stream.
     fn peek_is_boolop(&mut self) -> bool {
-        if let Symbol::BoolOp(_) = self.peek() {
-            true
-        } else {
-            false
-        }
+        matches!(self.peek(), Symbol::BoolOp(_))
```
Comment (sylvestre) [inline src/uu/test/src/parser.rs:124]: use of unstable library feature 'matches_macro'

Comment (jhscheer) [reply src/uu/test/src/parser.rs:124]: Oh.
Before I revert this, is a bump from Rust 1.40.0 to 1.42.0 planned any time soon?

Comment (sylvestre) [reply src/uu/test/src/parser.rs:124]: afaik, there isn't a driver for a rustc bump, sorry

File: src/uu/test/src/parser.rs [hunk 2]
```diff
@@ -121,6 +121,9 @@ impl Parser {
     /// Test if the next token in the stream is a BOOLOP (-a or -o), without
     /// removing the token from the stream.
     fn peek_is_boolop(&mut self) -> bool {
+        // matches!(self.peek(), Symbol::BoolOp(_))
+        #[allow(clippy::match_like_matches_macro)]
```
Comment (sylvestre) [inline src/uu/test/src/parser.rs:125]: attributes are not yet allowed on `if` expressions :)

Comment (jhscheer) [reply src/uu/test/src/parser.rs]: Hmm, to use attributes on 'if' expressions Rust 1.43 is needed... *sigh* I give up.

**Pull request merged**